config/index.ts

```diff
@@ -1,268 +1,406 @@
-  }
-  return config_DEPRECATED.loadConfig(path, options);
+import { MIN_HTTP_TIMEOUT } from '../constants/config';
+import { HubSpotConfigAccount } from '../types/Accounts';
+import { HubSpotConfig, ConfigFlag } from '../types/Config';
+import { CmsPublishMode } from '../types/Files';
+import { logger } from '../lib/logger';
+import {
+  getGlobalConfigFilePath,
+  getLocalConfigFilePath,
+  readConfigFile,
+  parseConfig,
+  buildConfigFromEnvironment,
+  writeConfigFile,
+  getLocalConfigFileDefaultPath,
+  getConfigAccountByIdentifier,
+  isConfigAccountValid,
+  getConfigAccountIndexById,
+  getConfigPathEnvironmentVariables,
+  getConfigAccountByInferredIdentifier,
+} from './utils';
+import { CMS_PUBLISH_MODE } from '../constants/files';
+import { Environment } from '../types/Config';
+import { i18n } from '../utils/lang';
+export function localConfigFileExists(): boolean {
+  return Boolean(getLocalConfigFilePath());
 }
 
-export function getAndLoadConfigIfNeeded(
-  options?: CLIOptions
-): Partial<CLIConfig> | null {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.config;
-  }
-  return config_DEPRECATED.getAndLoadConfigIfNeeded(options);
+export function globalConfigFileExists(): boolean {
+  return fs.existsSync(getGlobalConfigFilePath());
 }
 
-export function validateConfig(): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.validate();
-  }
-  return config_DEPRECATED.validateConfig();
-}
+function getDefaultConfigFilePath(): string {
+  const globalConfigFilePath = getGlobalConfigFilePath();
 
-export function loadConfigFromEnvironment(): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.useEnvConfig;
+  if (fs.existsSync(globalConfigFilePath)) {
+    return globalConfigFilePath;
   }
-  return Boolean(config_DEPRECATED.loadConfigFromEnvironment());
-}
 
-export function createEmptyConfigFile(
-  options: { path?: string } = {},
-  useHiddenConfig = false
-): void {
-  if (useHiddenConfig) {
-    CLIConfiguration.write({ accounts: [] });
-  } else {
-    return config_DEPRECATED.createEmptyConfigFile(options);
+  const localConfigFilePath = getLocalConfigFilePath();
+
+  if (!localConfigFilePath) {
+    throw new Error(i18n('config.getDefaultConfigFilePath.error'));
   }
+
+  return localConfigFilePath;
 }
 
-export function deleteEmptyConfigFile() {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.delete();
-  }
-  return config_DEPRECATED.deleteEmptyConfigFile();
+export function getConfigFilePath(): string {
+  const { configFilePathFromEnvironment } = getConfigPathEnvironmentVariables();
+
+  return configFilePathFromEnvironment || getDefaultConfigFilePath();
 }
 
-export function getConfig(): CLIConfig | null {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.config;
+export function getConfig(): HubSpotConfig {
+  const { useEnvironmentConfig } = getConfigPathEnvironmentVariables();
```

Comment:
Same with environment config - a user passes in the `--use-env` flag to the CLI, the CLI should set the `USE_ENVIRONMENT_CONFIG` env variable to tell LDL to use environment variables. These new environment variables let us avoid needing to have configPath and useEnv arguments for every function that accesses the config

Reply:
Good call, I can add something to the readme for these. On the CLI side, we'll be setting these automatically when users use `--use-env` or `--config` paths so nothing should change there